### PR TITLE
Change person search

### PR DESCRIPTION
**Pull request by AnnaGolosova**
## Link to ClickUp ticket

https://app.clickup.com/t/p9gw9g

### *Change /search/persons endpoint*

Adding new fields, update search for suggested changes

### *What changes have we introduced*

1. Add field TotalBalance to Person Model
2. Add fields TotalBalance, PostCode, PaymentReference to Tenure model from Person
3. Add field PersonType to GetPersonListRequest (Tenant, Leaseholder)
4. Add mapping extension for Person Types.
5. Change search: SearchText from Request is going on all fields, and one more filter for types.
6. Change tests for new search logic
7. Add README for first steps with ElasticSearch
8. Update PersonIndex

#### _Checklist_

- [x] Added end-to-end (i.e. integration) tests where necessary e.g to test complete functionality of a newly added feature
- [x] Added tests to cover all new production code
- [x] Added comments to the README or updated relevant documentation _(add link to documentation)_, where necessary.
- [x] Checked all code for possible refactoring
- [x] Code pipeline builds correctly

### CODE REVIEW

File: HousingSearchApi.Tests/V1/Helper/SearchPhraseTests.cs [hunk 1]
```diff
@@ -21,7 +24,21 @@ public SearchPhraseTests()
         [Theory]
         [InlineData("")]
         [InlineData(null)]
-        public void ShouldReturnNullIfRequestSearchTextIsEmpty(string searchText)
+        public void ShouldReturnFilterOnlyByTypeIfRequestSearchTextIsEmpty(string searchText)
```
Comment (evak6979) [inline HousingSearchApi.Tests/V1/Helper/SearchPhraseTests.cs:27]: Is there a reason why we no longer wish this to be null?

File: HousingSearchApi/V1/Interfaces/Sorting/IListSortFactory.cs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+using Nest;
```
Comment (evak6979) [inline HousingSearchApi/V1/Interfaces/Sorting/IListSortFactory.cs:1]: This interface does not appear to be used anywhere.

Comment (evak6979) [inline HousingSearchApi/V1/Interfaces/Sorting/IListSortFactory.cs:1]: This interface does not seem to be used.

**Review by evak6979 — COMMENTED** [review]
I would also add proper E2E steps ( In person steps and person stories ) . I think you've already updated the fixture as appropriate, but you still need to add steps + stories to cover the new functionality.

File: HousingSearchApi.Tests/V1/Helper/SearchPhraseTests.cs (hunk 1 above)
Comment (AnnaGolosova) [reply HousingSearchApi.Tests/V1/Helper/SearchPhraseTests.cs:27]: We'he changed search logic. Now we search by text and by type, and type is required. So if input text is empty - we will search only by type

File: HousingSearchApi/V1/Gateways/Models/QueryableIdentification.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
+using Nest;
+
+namespace HousingSearchApi.V1.Gateways.Models
```
Comment (LBHMGeorgieva) [inline HousingSearchApi/V1/Gateways/Models/QueryableIdentification.cs:3]: Hi Anna, why do we need this class? I know it's in the domain but we actually don't have the data migrated (and no plans to migrate it for now either)

File: HousingSearchApi/V1/Infrastructure/ElasticSearchExtensions.cs [hunk 4]
```diff
@@ -26,5 +27,29 @@ public static void ConfigureElasticSearch(this IServiceCollection services, ICon
 
             services.TryAddSingleton<IElasticClient>(esClient);
         }
+
+        /// <summary>
+        /// Applies Descending() or Ascending() method based on <paramref name="isDesc"/>
```
Comment (evak6979) [inline HousingSearchApi/V1/Infrastructure/ElasticSearchExtensions.cs:32]: There is another place to do this - check : https://github.com/LBHackney-IT/housing-search-api/blob/master/HousingSearchApi/V1/Interfaces/Sorting/SortFactory.cs . For person, should this still be as before, surname/firstname is already implemented. Alternatively add to what's already there. Thanks !

Comment (AnnaGolosova) [reply HousingSearchApi/V1/Infrastructure/ElasticSearchExtensions.cs:32]: This method is unused now. We can delete it

File: HousingSearchApi/V1/Gateways/Models/QueryableIdentification.cs (hunk 3 above)
Comment (AnnaGolosova) [reply HousingSearchApi/V1/Gateways/Models/QueryableIdentification.cs]: As I can see now, @evak6979 added almost same class 4 days ago, So I just removed this one from my PR
https://github.com/LBHackney-IT/housing-search-api/blob/master/HousingSearchApi/V1/Gateways/Models/Persons/QueryableIdentification.cs

File: HousingSearchApi/V1/Infrastructure/ElasticSearchExtensions.cs (hunk 4 above)
Comment (AnnaGolosova) [reply HousingSearchApi/V1/Infrastructure/ElasticSearchExtensions.cs]: Deleted

**Comment by AnnaGolosova** [comment]
@LBHMGeorgieva @evak6979 I fixed all your comments for this PR

**Comment by AnnaGolosova** [comment]
@LBHMGeorgieva @evak6979 I fixed all comments on this PR

**Review by evak6979 — DISMISSED** [review]

**Review by LBHMGeorgieva — APPROVED** [review]

**Comment by AnnaGolosova** [comment]
By discussion with Evangelos:
image

**Pull request merged**